vm_supervisor/network/hostnetwork.py

```diff
@@ -1,26 +1,144 @@
 import logging
+from ipaddress import IPv6Network
 from pathlib import Path
+from typing import Optional, Protocol
+
+from aleph_message.models import ItemHash
+
+from vm_supervisor.conf import IPv6AllocationPolicy
 
 from .firewall import initialize_nftables, setup_nftables_for_vm, teardown_nftables
 from .interfaces import TapInterface
 from .ipaddresses import IPv4NetworkWithInterfaces
+from .ndp_proxy import NdpProxy
 
 logger = logging.getLogger(__name__)
 
 
+def _read_proc_file_value(config_file: Path) -> int:
+    return int(config_file.read_text())
+
+
 def get_ipv4_forwarding_state() -> int:
-    """Reads the current ipv4 forwarding setting from the hosts, converts it to int and returns it"""
-    return int(Path("/proc/sys/net/ipv4/ip_forward").read_text())
+    """Reads the current IPv4 forwarding setting from the host, converts it to int and returns it"""
+    return _read_proc_file_value(Path("/proc/sys/net/ipv4/ip_forward"))
+
+
+def get_ipv6_forwarding_state() -> int:
```

Comment:
Since that file can only contain 0 or 1, should we use booleans instead ?

Reply:
We might, but then we have to convert between booleans and integers. I kept the previous behaviour, feel free to change it in another PR.